airflow/decorators/base.py

```diff
@@ -29,7 +29,60 @@
 from airflow.utils.task_group import TaskGroup, TaskGroupContext
 
 
-class BaseDecoratedOperator(BaseOperator):
+def validate_python_callable(python_callable):
+    """
+    Validate that python callable can be wrapped by operator.
+    Raises exception if invalid.
+
+    :param python_callable: Python object to be validated
+    :raises: TypeError, AirflowException
+    """
+    if not callable(python_callable):
+        raise TypeError('`python_callable` param must be callable')
+    if 'self' in signature(python_callable).parameters.keys():
+        raise AirflowException('@task does not support methods')
```

Comment:
What about class methods?

Reply:
@turbaszek what do you mean? I imagine we want to keep parity with how the task decorator currently works